x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/EsqlBinaryComparison.java

```diff
@@ -118,6 +125,26 @@ protected EsqlBinaryComparison(
         this.functionType = operation;
     }
 
+    public static EsqlBinaryComparison readFrom(StreamInput in) throws IOException {
+        // TODO this uses a constructor on the operation *and* a name which is confusing. It only needs one. Everything else uses a name.
+        var source = Source.readFrom((PlanStreamInput) in);
+        EsqlBinaryComparison.BinaryComparisonOperation operation = EsqlBinaryComparison.BinaryComparisonOperation.readFromStream(in);
+        var left = ((PlanStreamInput) in).readExpression();
+        var right = ((PlanStreamInput) in).readExpression();
+        // TODO: Remove zoneId entirely
+        var zoneId = in.readOptionalZoneId();
+        return operation.buildNewInstance(source, left, right);
+    }
+
+    @Override
+    public final void writeTo(StreamOutput out) throws IOException {
+        source().writeTo(out);
```

Comment:
Nit: would be neat if we could delegate the source writing to a superclass, but not sure if the complexity would be worth it.